### PR TITLE
Adapt to lts-8.6 and use proto-lens-0.2.0.1

### PR DESCRIPTION
**Pull request by jcberentsen**
Squashed previous pull request

### CODE REVIEW

**Comment by judah** [comment]
Thanks for the patch.  Can you please confirm that it builds warning-free with `stack clean; stack build --ghc-options='-Wall -Werror'`?

In particular, while testing out the proto-related changes just now I noticed that you'll need to fix a typo of mine:

    --- a/tensorflow/src/TensorFlow/Types.hs
    +++ b/tensorflow/src/TensorFlow/Types.hs
    @@ -410,7 +410,7 @@ instance All Eq (Map f as) => Eq (ListOf f as) where
         (x :/ xs) == (y :/ ys) = x == y && xs == ys
         -- Newer versions of GHC use the GADT to tell that the previous cases are
         -- exhaustive.
    -#if _GLASGOW_HASKELL__ < 800
    +#if __GLASGOW_HASKELL__ < 800
         _ == _ = False
     #endif

**Comment by jcberentsen** [comment]
I did notice that warning at first build. I should have looked closer. I'll
fix.

Den tir. 11. apr. 2017, 10:24 skrev Judah Jacobson <notifications@github.com

File: tensorflow/src/TensorFlow/Session.hs [hunk 1]
```diff
@@ -135,7 +135,7 @@ extend = do
     trace <- Session (asks tracer)
     nodesToExtend <- build flushNodeBuffer
     unless (null nodesToExtend) $ liftIO $ do
-        let graphDef = def & node .~ nodesToExtend
+        let graphDef = (def :: GraphDef) & node .~ nodesToExtend
```
Comment (judah) [inline tensorflow/src/TensorFlow/Session.hs]: For future reference (no changes needed for this PR): `proto-lens-0.2.*` made this extra type annotation necessary.  However, we can avoid it by switching the tensorflow packages away from `lens-family`.  Any of `lens-family-core`, `microlens` or `lens` would be sufficient.

Some notes on this topic are here:
https://github.com/google/proto-lens/blob/master/proto-lens-tests/tests/labels_test.hs#L40

**Comment by judah** [comment]
Jenkins, test this please.

**Comment by judah** [comment]
LGTM

**Pull request merged**